src/pages/iou/IOUDetailsModal.js

```diff
@@ -77,29 +81,109 @@ class IOUDetailsModal extends Component {
     constructor(props) {
         super(props);
 
+        // We always have the option to settle manually
+        const paymentOptions = [CONST.IOU.PAYMENT_TYPE.ELSEWHERE];
+
+        // Only allow settling via PayPal.me if the submitter has a name set
+        if (props.iouReport.submitterPayPalMeAddress) {
+            paymentOptions.push(CONST.IOU.PAYMENT_TYPE.PAYPAL_ME);
+        }
+
+        this.submitterPhoneNumber = undefined;
+        this.isComponentMounted = false;
+
         this.state = {
-            // Temporarily placeholder, as the Venmo/Paypal issue will introduce settlement types.
-            settlementType: 'Elsewhere',
+            paymentType: CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
+            isSettlementMenuVisible: false,
+            paymentOptions,
         };
 
         this.performIOUPayment = this.performIOUPayment.bind(this);
     }
 
     componentDidMount() {
+        this.isComponentMounted = true;
         fetchIOUReportByID(this.props.route.params.iouReportID, this.props.route.params.chatReportID);
+        this.addVenmoPaymentOptionIfAvailable();
+    }
+
+    componentWillUnmount() {
+        this.isComponentMounted = false;
+    }
+
+    setMenuVisibility(isSettlementMenuVisible) {
+        this.setState({isSettlementMenuVisible});
     }
 
     performIOUPayment() {
         payIOUReport({
             chatReportID: this.props.route.params.chatReportID,
             reportID: this.props.route.params.iouReportID,
-            paymentMethodType: this.state.settlementType,
+            paymentMethodType: this.state.paymentType,
+            amount: this.props.iouReport.total,
+            currency: this.props.iouReport.currency,
+            submitterPayPalMeAddress: this.props.iouReport.submitterPayPalMeAddress,
+            submitterPhoneNumber: this.submitterPhoneNumber,
         });
     }
 
+    /**
+     * @param {String} phoneNumber
+     * @returns {Boolean}
+     */
+    isValidUSPhone(phoneNumber) {
+        // Remove alphanumeric characters and validate that this is in fact a phone number
+        return CONST.REGEX.PHONE_E164_PLUS.test(phoneNumber.replace(CONST.REGEX.NON_ALPHA_NUMERIC, ''))
+
+            // Next make sure it's a US phone number
+            && CONST.REGEX.US_PHONE.test(phoneNumber);
+    }
+
```

Comment:
NAB: I guess we can wait until it's reused, but this doesn't seem the best place for this logic.

Reply:
Had a similar thought, but we might never re-use it so I think it's a fine place for it until we need to.

Reply:
Makes sense